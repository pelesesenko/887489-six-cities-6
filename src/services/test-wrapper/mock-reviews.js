export const mockReviews = [
  {
    id: 1,
    user: {
      id: 18,
      isPro: true,
      name: `Sophie`,
      avatarUrl:
        `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
    },
    rating: 5,
    comment:
      `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
    date: `2021-02-23T08:04:28.647Z`,
  },
  {
    id: 2,
    user: {
      id: 15,
      isPro: false,
      name: `Kendall`,
      avatarUrl:
        `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    },
    rating: 3,
    comment:
      `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2021-02-20T08:04:28.647Z`,
  },
];
