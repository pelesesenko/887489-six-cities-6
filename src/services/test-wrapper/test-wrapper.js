import React from 'react';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import * as states from './mock-states';

export const history = createMemoryHistory();
export const TestWrapper = ({
  children,
  init = false,
  url = null,
  stateModifier = {},
  mockDispatch = () =>{},
  mockHistory = null,
}) => {

  const mockStore = configureStore({});
  const store = mockStore(() => ({
    ...init ? states.initial : states.filled,
    ...stateModifier,
  }));

  if (!mockHistory) {
    mockHistory = createMemoryHistory();
  }

  if (url) {
    mockHistory.push(url);
  }
  store.dispatch = mockDispatch;

  return (
    <Provider store={store}>
      <Router history={mockHistory}>
        {children}
      </Router>
    </Provider>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.arrayOf(PropTypes.object)]).isRequired,
  init: PropTypes.bool,
  url: PropTypes.string,
  stateModifier: PropTypes.object,
  mockDispatch: PropTypes.func,
  mockHistory: PropTypes.object,
};
