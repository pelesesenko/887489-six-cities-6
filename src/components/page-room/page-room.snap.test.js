import React from 'react';
import {render} from '@testing-library/react';
import {TestWrapper} from '../../services/test-wrapper/test-wrapper';
import {mockReviews} from '../../services/test-wrapper/mock-reviews';

import PageRoom from './page-room';


it(`PageRoom should render correctly`, () => {

  const mockSetReviews = jest.fn();
  jest.spyOn(React, `useState`).mockImplementation(() => [mockReviews, mockSetReviews]);
  jest.spyOn(React, `useEffect`).mockImplementation(() => {});

  const {container} = render(
      <TestWrapper >
        <PageRoom id={3} isAuthorized={true} isRoomLoaded={true} isNearbyLoaded={true}/>
      </TestWrapper>
  );

  expect(container).toMatchSnapshot();
});
