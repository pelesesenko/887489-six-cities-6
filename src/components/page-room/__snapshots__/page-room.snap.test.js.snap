// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageRoom should render correctly 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    123@mail.ru
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <button
                  class="button"
                  style="margin-left: 30px;"
                >
                  <svg
                    fill="none"
                    height="22"
                    stroke="#777"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="22"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                    />
                    <polyline
                      points="16 17 21 12 16 7"
                    />
                    <line
                      x1="21"
                      x2="9"
                      y1="12"
                      y2="12"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--property"
    >
      <section
        class="property"
      >
        <div
          class="property__gallery-container container"
        >
          <div
            class="property__gallery"
          >
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
              />
            </div>
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                class="property__image"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="property__container container"
        >
          <div
            class="property__wrapper"
          >
            <div
              class="property__name-wrapper"
            >
              <h1
                class="property__name"
              >
                Amazing and Extremely Central Flat
              </h1>
              <button
                class="property__bookmark-button button
                    null"
                type="button"
              >
                <svg
                  class="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="property__rating rating"
            >
              <div
                class="property__stars rating__stars"
              >
                <span
                  style="width: 40%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                class="property__rating-value rating__value"
              >
                2.3
              </span>
            </div>
            <ul
              class="property__features"
            >
              <li
                class="property__feature property__feature--entire"
              >
                Room
              </li>
              <li
                class="property__feature property__feature--bedrooms"
              >
                1
                 Bedrooms
              </li>
              <li
                class="property__feature property__feature--adults"
              >
                Max 
                1
                 adults
              </li>
            </ul>
            <div
              class="property__price"
            >
              <b
                class="property__price-value"
              >
                €
                145
              </b>
              <span
                class="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              class="property__inside"
            >
              <h2
                class="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                class="property__inside-list"
              >
                <li
                  class="property__inside-item"
                >
                  Laptop friendly workspace
                </li>
                <li
                  class="property__inside-item"
                >
                  Breakfast
                </li>
              </ul>
            </div>
            <div
              class="property__host"
            >
              <h2
                class="property__host-title"
              >
                Meet the host
              </h2>
              <div
                class="property__host-user user"
              >
                <div
                  class="property__avatar-wrapper user__avatar-wrapper
                      property__avatar-wrapper--pro"
                >
                  <img
                    alt="Host avatar"
                    class="property__avatar user__avatar"
                    height="74"
                    src="img/avatar-angelina.jpg"
                    width="74"
                  />
                </div>
                <span
                  class="property__user-name"
                >
                  Angelina
                </span>
              </div>
              <div
                class="property__description"
              >
                <p
                  class="property__text"
                >
                  A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.
                </p>
              </div>
            </div>
            <section
              class="property__reviews reviews"
            >
              <h2
                class="reviews__title"
              >
                Reviews · 
                <span
                  class="reviews__amount"
                >
                  2
                </span>
              </h2>
              <ul
                class="reviews__list"
              >
                <li
                  class="reviews__item"
                >
                  <div
                    class="reviews__user user"
                  >
                    <div
                      class="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        class="reviews__avatar user__avatar"
                        height="54"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
                        width="54"
                      />
                    </div>
                    <span
                      class="reviews__user-name"
                    >
                      Sophie
                    </span>
                  </div>
                  <div
                    class="reviews__info"
                  >
                    <div
                      class="reviews__rating rating"
                    >
                      <div
                        class="reviews__stars rating__stars"
                      >
                        <span
                          style="width: 100%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      class="reviews__text"
                    >
                      The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
                    </p>
                    <time
                      class="reviews__time"
                      datetime="02/23/2021"
                    >
                      February 2021
                    </time>
                  </div>
                </li>
                <li
                  class="reviews__item"
                >
                  <div
                    class="reviews__user user"
                  >
                    <div
                      class="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        class="reviews__avatar user__avatar"
                        height="54"
                        src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg"
                        width="54"
                      />
                    </div>
                    <span
                      class="reviews__user-name"
                    >
                      Kendall
                    </span>
                  </div>
                  <div
                    class="reviews__info"
                  >
                    <div
                      class="reviews__rating rating"
                    >
                      <div
                        class="reviews__stars rating__stars"
                      >
                        <span
                          style="width: 60%;"
                        />
                        <span
                          class="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      class="reviews__text"
                    >
                      The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.
                    </p>
                    <time
                      class="reviews__time"
                      datetime="02/20/2021"
                    >
                      February 2021
                    </time>
                  </div>
                </li>
              </ul>
              <form
                action="#"
                class="reviews__form form"
                method="post"
              >
                <label
                  class="reviews__label form__label"
                  for="review"
                >
                  Your review
                </label>
                <div
                  class="reviews__rating-form form__rating"
                >
                  <input
                    class="form__rating-input visually-hidden"
                    id="5-stars"
                    name="rating"
                    type="radio"
                    value="5"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="5-stars"
                    title="perfect"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="4-stars"
                    name="rating"
                    type="radio"
                    value="4"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="4-stars"
                    title="good"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="3-stars"
                    name="rating"
                    type="radio"
                    value="3"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="3-stars"
                    title="not bad"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="2-stars"
                    name="rating"
                    type="radio"
                    value="2"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="2-stars"
                    title="badly"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="1-stars"
                    name="rating"
                    type="radio"
                    value="1"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="1-stars"
                    title="terribly"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                </div>
                <textarea
                  class="reviews__textarea form__textarea"
                  id="review"
                  name="comment"
                  placeholder="Tell how was your stay, what you like and what can be improved"
                />
                <div
                  class="reviews__button-wrapper"
                >
                  <p
                    class="reviews__help"
                  >
                    To submit review please make sure to set 
                    <span
                      class="reviews__star"
                    >
                      rating
                    </span>
                     and describe your stay with at least 
                    <b
                      class="reviews__text-amount"
                    >
                      50 characters
                    </b>
                    .
                  </p>
                  <button
                    class="reviews__submit form__submit button"
                    disabled=""
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <section
          class="property__map map"
        >
          <div
            class="leaflet-container leaflet-grab leaflet-touch-drag"
            id="map"
            style="height: 579px; width: 1144px; margin: 0px auto; position: relative;"
            tabindex="0"
          >
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                      style="width: 256px; height: 256px; left: -130px; top: -72px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="./img/pin.svg"
                  style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 76px; top: 195px; z-index: 195;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="./img/pin.svg"
                  style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 134px; top: 88px; z-index: 88;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="./img/pin.svg"
                  style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 76px; top: 168px; z-index: 168;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="./img/pin-active.svg"
                  style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 14777px; top: -32261px; z-index: -32261;"
                  tabindex="0"
                />
              </div>
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              />
              <div
                class="leaflet-top leaflet-right"
              />
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | © 
                  <a
                    href="https://www.openstreetmap.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                   contributors © 
                  <a
                    href="https://carto.com/attributions"
                  >
                    CARTO
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
      </section>
      <div
        class="container"
      >
        <section
          class="near-places places"
        >
          <h2
            class="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            class="near-places__list places__list"
          >
            <article
              class="place-card near-places__card"
            >
              <div
                class="place-card__image-wrapper near-places__image-wrapper"
              >
                <a
                  href="/offer/13"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                class="place-card__info "
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      346
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                       / night
                    </span>
                  </div>
                  <button
                    class="place-card__bookmark-button button place-card__bookmark-button--active"
                    type="button"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      In bookmarks
                    </span>
                  </button>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 40%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/13"
                  >
                    The Joshua Tree House
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  Hotel
                </p>
              </div>
            </article>
            <article
              class="place-card near-places__card"
            >
              <div
                class="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                class="place-card__image-wrapper near-places__image-wrapper"
              >
                <a
                  href="/offer/23"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                class="place-card__info "
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      558
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                       / night
                    </span>
                  </div>
                  <button
                    class="place-card__bookmark-button button false"
                    type="button"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 60%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/23"
                  >
                    Loft Studio in the Central Area
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  House
                </p>
              </div>
            </article>
            <article
              class="place-card near-places__card"
            >
              <div
                class="place-card__image-wrapper near-places__image-wrapper"
              >
                <a
                  href="/offer/76"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                class="place-card__info "
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      259
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                       / night
                    </span>
                  </div>
                  <button
                    class="place-card__bookmark-button button place-card__bookmark-button--active"
                    type="button"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      In bookmarks
                    </span>
                  </button>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 60%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/76"
                  >
                    Penthouse, 4-5 rooms + 5 balconies
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  Room
                </p>
              </div>
            </article>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;
